simplify rank series by source

nemozen opened the pull request:
we remove teh bit that deals with duplicates in the list of series. this is over-optimization.
the caller can easily avoid duplicates, and if not, it's relatively harmless in ranking

nemozen committed:
simplify rank series by source

we remove teh bit that deals with duplicates in the list of series. this is over-optimization.
the caller can easily avoid duplicates, and if not, it's relatively harmless in ranking

nemozen committed:
incremement version number

muzigao commented on api/client/lib.py:
```diff
@@ -334,15 +334,7 @@ def make_key(key):
-        key=lambda x: x[0])) for single_series in series_list)
-
-    for series in map(dict, selections_sorted):
+    for series in series_list:
```
need to update test_rank_series_by_source

nemozen committed:
update unit test

itsjohnward commented on api/client/lib.py:
```diff
@@ -334,15 +334,7 @@ def make_key(key):
-    # returns a different order for the same dictionary. See test case.
-    selections_sorted = set(tuple(sorted(
-        [k_v for k_v in iter(list(single_series.items()))
-         if k_v[0] not in ('source_id', 'source_name')],
```
Now we've got that source_id-source_name not matching behavior again. Need to filter out source_name, or else if you pass in a series object like `{..., 'source_id': 2, 'source_name': 'FAO'}` from `get_data_series()` it'll return something like

```py
[ {..., 'source_id': 2, 'source_name': 'FAO'},
  {..., 'source_id': 14, 'source_name': 'FAO'},
  {..., 'source_id': 26, 'source_name': 'FAO'} ]
```

itsjohnward replied:
^ Good thing to add a unit test for as well

itsjohnward replied:
ref https://github.com/gro-intelligence/api-client/pull/125/commits/04a449523641783440035464c064034270ba9572

nemozen replied:
Done. Cleaned up so pre-selected source is ignored since the goal is to consider all possible sources

nemozen committed:
update second test assertion

nemozen committed:
remove pre-selected source if any in rank_series_by_source input

we want to consider *all* sources for the given selection, so prior selection of source can be ignored

itsjohnward reviewed (approved):
Tested with

```py
from api.client.gro_client import GroClient
import os
client = GroClient('api.gro-intelligence.com', os.environ['GROAPI_TOKEN'])

for ranking in client.rank_series_by_source([{'metric_id': 860032, 'item_id': 274, 'region_id': 1215}]):
  print('ranking 1', ranking)

for ranking in client.rank_series_by_source([{'metric_id': 860032, 'item_id': 274, 'region_id': 1215, 'source_id': 2}]):
  print('ranking with source', ranking)

for ranking in client.rank_series_by_source([{'metric_id': 860032, 'item_id': 274, 'region_id': 1215, 'start_date': '2012-01-01'}]):
  print('ranking with date', ranking)

for data_series in client.find_data_series(item='corn', region='united states'):
  print('data_series', data_series)
```

Seeing source_name is excluded now and it works with historical regions metadata. LGTM
